core/controllers/base_test.py

```diff
@@ -107,9 +120,55 @@ def test_redirect_in_both_logged_in_and_logged_out_states(self):
         self.login('user@example.com')
         response = self.testapp.get('/')
         self.assertEqual(response.status_int, 302)
+
+        # Homepage is either dashboard or library based on user's learner
+        # status, this homepage redirect rule behavior is seperately tested
+        atHome = ('dashboard' in response.headers['location'] or
```

Comment:
@seanlip  @jaredsilver 
A new use case came to my mind. According to current approach users who have created an exploration i past are creators. But what if a user just edited another user's created explorations and not created any exploration ? Will that user be treated as a creator and be redirected to dashboard ?